packages/app-admin-ui/client/pages/List/index.js

```diff
@@ -34,22 +35,10 @@ const HeaderInset = props => (
   <div css={{ paddingLeft: gridSize * 2, paddingRight: gridSize * 2 }} {...props} />
 );
 
-type Props = {
-  adminMeta: Object,
-  list: Object,
-  routeProps: Object,
-};
-type LayoutProps = Props & {
-  items: Array<Object>,
-  itemCount: number,
-  queryErrors: Array<Object>,
-};
-
-export function ListLayout(props: LayoutProps) {
-  const { adminMeta, items, itemCount, queryErrors, list, routeProps, query } = props;
-  const [showCreateModal, toggleCreateModal] = useState(false);
+export function ListLayout(props) {
+  const { adminMeta, items, itemCount, queryErrors, routeProps, query } = props;
   const measureElementRef = useRef();
-
+  const { list, openCreateItemModal } = useList();

```

Comment:
I'm not a fan of the `openCreateItemModal` thing because I'm not sure how to answer the question of "What modal will open when I call this function?" If I render two modals, will they both open? Will it open modals that are in totally different parts of the react tree

Reply:
It only open the modal inside the list provider. I think the pattern can be improved but I don't love unrelated wrapper components controlling the state of the modal by passing props around. This really creates a dis-connect with the modal and it's state. 

I think perhaps the right pattern is to either import the modal from the list provider so it's clear there is one instance of a modal per list provider. 

An alternative is to create a modal provider.

Reply:
should there be a `useItemModal({...listConfig})` which let us use modals when needed, we would also need some models like `DeleteItemModal` `DeleteManyItemsModel`, `updateManyItemsModel` and perhaps a new `updateItemModel` (when we really have a list item detail page rather than a pure editing experience)

Reply:
Yep, I want to move towards something like that but I consider this a step on that path.